### ISSUE TITLE
Rethinking Spatial Dimensions of Vision Transformers

### ISSUE THREAD

**Issue by rentainhe** [rentainhe/what_I_have_read#3]
## Papers
[Rethinking Spatial Dimensions of Vision Transformers](https://arxiv.org/abs/2103.16302)

## Overview
![image](https://user-images.githubusercontent.com/48727989/127746096-13ea8467-2f0e-4fb4-8bca-3afaf280f953.png)

## Motivation
- Effective architecture for ViT
- Revisit the role of **spatial dimension reduction** in ViT which follows the rule of CNN based Model
- Design a pyramid architecture for vision transformers

### Effect of pooling in CNN
**ResNet50 with pooling** vs **ResNet50 without pooling**
Settings: 
- only change the stem in **ResNet50 without pooling** to get the same feature size after CNN stem
![image](https://user-images.githubusercontent.com/48727989/127746335-9ac935c7-6c2c-492f-b3ad-8514412ecf4c.png)

pooling layer benefits:
- increase the capability of architecture as it has lower training loss
- increase the model's generalization performance

### Pooling-based Vision Transformer(PiT)
![image](https://user-images.githubusercontent.com/48727989/127746493-4d8b843d-d27b-46c3-bc78-4f404e10ddf7.png)

Details:
- depth-wise convolution for features
- use a linear projection for **class token** or **distillation token**

### Effects of the pooling layer in ViT
![image](https://user-images.githubusercontent.com/48727989/127746568-ae175f1a-a438-4fe7-aa8e-512e88c8c64a.png)

## Experiments
Three Settings:
- AdamP optimizer

First: Vanilla settings for 100 epoches
Second: CutMix settings for 100 epoches
Third: DeiT settings for 300 epoches
